### PR TITLE
Fixed a couple license headers on files and updated README

### LINKED ISSUE #59: Provide initial set of documentation in the README.md
Right now our README is sparse; it provides how to run unit tests and pep8 but nothing about what Bakula is or how to use it outside of running our unit tests. This should be done prior to Strata + Hadoop World.

### PR DESCRIPTION
**Pull request by ericjperry**
Updated the README with more information about Bakula and updated a few license headers to Apache 2.

@KrisSiegel @sapanshah let me know what you guys think.

Resolves #59.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,67 @@
 ## Bakula
 
+Bakula is a reactive programming framework which allows developers to quickly and
```
Comment (KrisSiegel) [inline README.md]: I would argue that Bakula is more of a platform than a framework. But I think this is subjective enough that you could get different opinions from many different people :)

File: README.md [hunk 2]
```diff
@@ -1,5 +1,67 @@
...
+easily build event-driven applications with no strict API to implement. Bakula
+events are propagated via registratered 'topics' to user-provided Docker
+containers for processing.
+
+Bakula can be used with any Docker Registry (including the Docker Hub) to
+pull Docker images and spin up containers for processing events.
+
+### Events
+
+An 'event' in Bakula is any data provided to the service via POSTing to the
```
Comment (KrisSiegel) [inline README.md]: While an event can be anything it's essentially expecting a file to be uploaded as the data itself. So we may want to be a little more explicit about the additional fields needing to be files.

**Comment by KrisSiegel** [comment]
Pretty good :+1:

File: README.md (hunk 2 above)
Comment (ericjperry) [inline README.md]: Added another sentence for clarity. Yea or nay?

File: README.md (hunk 1 above)
Comment (ericjperry) [inline README.md]: Changed to platform.

File: README.md (hunk 2 above)
Comment (KrisSiegel) [inline README.md]: Sounds good

File: README.md [hunk 3]
```diff
@@ -1,5 +1,68 @@
...
+When events enter the system, Bakula uses registrations to determine which containers
+should be run, and when. Each Bakula registration contains the following:
+
+* Topic: A topic is simply a string that identifies a type of event. For example a
+stream of twitter data may be sent to the 'Twitter' topic.
+* Container: The full name of the Docker image to be pulled and started as a container
+when events occur for this topic.
+* Threshold (optional): The number of events which must occur before a container is
+started. For example, a threshold of 1000 would denote that 1000 events must occur before
+a container is created for this registration.
```
Comment (sapanshah) [inline README.md]: instead of created say 'started'

**Comment by sapanshah** [comment]
Other than my minor nitpick :+1:

**Comment by KrisSiegel** [comment]
Need to make sure this gets pulled into the other repo

File: README.md (hunk 3 above)
Comment (ericjperry) [inline README.md]: fixed

**Pull request merged**